libsolidity/analysis/TypeChecker.cpp

```diff
@@ -346,7 +346,7 @@ void TypeChecker::checkContractExternalTypeClashes(ContractDefinition const& _co
 		for (size_t i = 0; i < it.second.size(); ++i)
 			for (size_t j = i + 1; j < it.second.size(); ++j)
 				if (!it.second[i].second->hasEqualArgumentTypes(*it.second[j].second))
-					m_errorReporter.fatalTypeError(
+					m_errorReporter.typeError(
```

Comment:
Please either leave this fatal or try to crash the compiler in such a situation :-)